### PR TITLE
feat: added possibility to payout to connected account

### PR DESCRIPTION
**Pull request by vackca**

### CODE REVIEW

File: microservices/payment/src/services/payment-gateway/stripe.ts [hunk 1]
```diff
@@ -247,31 +273,100 @@ class Stripe extends Abstract {
...
+      status: TransactionStatus.INITIAL,
+    });
+
+    await this.transactionRepository.save(transaction);
+  }
+
+  /**
+   * Creates payout transfers for given entities
+   */
+  public payout(entitiesId: { id: string; userId: string }[]) {
```
Comment (MatthewPattell) [inline microservices/payment/src/services/payment-gateway/stripe.ts]:
```suggestion
  public payout(entitiesIds: { id: string; userId: string }[]) {
```

File: microservices/payment/src/services/payment-gateway/stripe.ts [hunk 2]
```diff
@@ -247,31 +273,100 @@ class Stripe extends Abstract {
...
+    entitiesId.forEach(({ id, userId }) => {
+      void this.createTransfer(id, userId);
+    });
```
Comment (MatthewPattell) [inline microservices/payment/src/services/payment-gateway/stripe.ts]: you need use `for in` and `await Promise.all or allSetted`

File: microservices/payment/src/methods/stripe/payout.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    const { paymentProvider } = await remoteConfig();
+
+    if (paymentProvider !== PaymentProvider.STRIPE) {
+      throw new Error('Payout only suitable for the stripe payment provider');
+    }
+
+    const service = (await Factory.create(getManager())) as Stripe;
+
+    return {
+      isHandled: service.payout(entities),
```
Comment (MatthewPattell) [inline microservices/payment/src/methods/stripe/payout.ts]:
```suggestion
      isComplete: service.payout(entities),
```

File: microservices/payment/src/interfaces/remote-config.ts [hunk 4]
```diff
@@ -8,4 +8,5 @@ export interface IRemoteConfig {
   paymentProvider?: PaymentProvider;
   paymentOptions?: TPaymentOptions;
   paymentWebhookKey: string;
+  payoutCoeff: number;
```
Comment (MatthewPattell) [inline microservices/payment/src/interfaces/remote-config.ts]:
```suggestion
  paymentWebhookKey?: string;
  payoutCoeff?: number;
```

**Review by MatthewPattell — CHANGES_REQUESTED** [review]

File: microservices/payment/src/services/payment-gateway/stripe.ts (hunk 2 above)
Comment (vackca) [reply microservices/payment/src/services/payment-gateway/stripe.ts]: Can i use map for this? Check please, can I use it hte way I fixed it?

File: microservices/payment/src/services/payment-gateway/stripe.ts [hunk 5]
```diff
@@ -247,31 +273,108 @@ class Stripe extends Abstract {
...
+
+  /**
+   * Creates payout transfers for given entities
+   */
+  public async payout(entitiesIds: { id: string; userId: string }[]) {
+    const { payoutCoeff } = await remoteConfig();
+
+    // TODO: create mechanism to rearrange/mark payouts with errors to deal with them later
+    if (!payoutCoeff) {
+      Log.error(`Payout coefficient is not provided`);
```
Comment (MatthewPattell) [inline microservices/payment/src/services/payment-gateway/stripe.ts]:
```suggestion
      Log.error('Payout coefficient is not provided');
```

**Review by MatthewPattell — APPROVED** [review]

**Pull request merged**